### ISSUE TITLE
[Studio] Wire dLLM training backend — integrate DiffusionTrainer in worker subprocess

### ISSUE THREAD

**Issue by nishide-dev** [nishide-dev/unturtle#154]
## Background / Goal

The Studio training subprocess currently uses `UnslothTrainer` (wrapping `SFTTrainer` from TRL). For dLLM models (a2d, llada, dream), the subprocess must instead use `FastDiffusionModel` + `DiffusionTrainer` + `DiffusionTrainingArguments` from `unturtle`.

## Architecture

Rather than rewriting `core/training/trainer.py`, add a new `DiffusionTrainerBackend` class that mirrors the existing `UnslothTrainer` interface. The worker subprocess branches based on `config["model_type"]`.

```
worker.py
  if config["model_type"] in ["a2d", "llada", "dream"]:
      backend = DiffusionTrainerBackend(config, event_queue)
  else:
      backend = UnslothTrainer(config, event_queue)   # unchanged
  backend.start_training()
```

## Acceptance Criteria

- [ ] `DiffusionTrainerBackend` implements the same interface as `UnslothTrainer`: `load_model()`, `load_and_format_dataset()`, `start_training()`
- [ ] `TrainingStartRequest` Pydantic model accepts `alpha_scheduler`, `time_epsilon`, `loss_weight_type`, `completion_only`, `model_type`
- [ ] A training job launched with `model_type: "llada"` calls `FastDiffusionModel.from_pretrained()` and `DiffusionTrainer` — verified via log output
- [ ] Progress events (`step`, `loss`, `epoch`) stream back through the existing SSE endpoint identically to causal LM training
- [ ] The existing causal LM training path (model_type not in dLLM list) is **unaffected**
- [ ] If `unturtle` is not importable in the subprocess, a clear error event is emitted (not a silent crash)

## DiffusionTrainerBackend Design

```python
# unturtle_studio/backend/core/training/diffusion_trainer.py

class DiffusionTrainerBackend:
    def __init__(self, config: dict, event_queue):
        self.config = config
        self.event_queue = event_queue

    def load_model(self):
        from unturtle import FastDiffusionModel
        model, tokenizer = FastDiffusionModel.from_pretrained(
            model_name=self.config["model"],
            max_seq_length=self.config.get("max_seq_length", 2048),
            load_in_4bit=self.config.get("load_in_4bit", False),
        )
        if self.config.get("use_lora"):
            model = FastDiffusionModel.get_peft_model(model, **lora_kwargs)
        self.model, self.tokenizer = model, tokenizer

    def load_and_format_dataset(self): ...

    def start_training(self):
        from unturtle.diffusion.trainer import DiffusionTrainer, DiffusionTrainingArguments
        args = DiffusionTrainingArguments(
            alpha_scheduler=self.config.get("alpha_scheduler", "linear"),
            time_epsilon=self.config.get("time_epsilon", 1e-3),
            loss_weight_type=self.config.get("loss_weight_type", "uniform"),
            completion_only=self.config.get("completion_only", False),
            ...
        )
        trainer = DiffusionTrainer(model=self.model, args=args, ...)
        trainer.train()
```

## Key Files to Touch

| File | Change |
|------|--------|
| `unturtle_studio/backend/core/training/diffusion_trainer.py` | **New file** — `DiffusionTrainerBackend` class |
| `unturtle_studio/backend/core/training/worker.py` | Branch on `config["model_type"]` to select backend |
| `unturtle_studio/backend/models/training.py` | Add `alpha_scheduler`, `time_epsilon`, `loss_weight_type`, `completion_only`, `model_type` to `TrainingStartRequest` |

## References

- `unturtle/diffusion/trainer.py` — `DiffusionTrainer`, `DiffusionTrainingArguments`
- `unturtle/fast_diffusion_model.py` — `FastDiffusionModel.from_pretrained`, `get_peft_model`
- `unturtle_cli/commands/train.py` — existing CLI train path as reference for the API call sequence

## Dependencies

- #147, #152 (Studio must be launchable to test end-to-end)

**Comment by nishide-dev** [comment]
Addressed in merged PRs (#164, #165, #166).
